### PR TITLE
feat: add data attributes definition for learning subdomain

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
**Description:** 
This PR adds class definitions for the objects that will be sent by Open edX events, they are based in the library

**JIRA:**
https://edunext.atlassian.net/browse/PS2021-629?atlOrigin=eyJpIjoiODIwNDA5M2MxZGRlNGQ2NjhjNjJhZWIwNWZiNWNlYjUiLCJwIjoiaiJ9

**Dependencies:** 
This PR depends on #7. Merged already

**Testing instructions:**
In your Open edX devstack
1. Go to lms shell `make lms-shell`
2. Install openedx-events specifying the events definition branch:
 `pip install git+https://github.com/eduNEXT/openedx-events.git@MJG/event_data#egg=event_data` 
3. Go to the lms Django shell: `python manage.py lms shell`
4. Now, you can import the platform objects and map them to attrs classes. For example, you could do:

```
>>> from common.djangoapps.student.models import CourseEnrollment
>>> from openedx_events.learning.data import CourseEnrollmentData, UserData, UserPersonalData, CourseOverviewData
>>> course_enrollment = CourseEnrollment.objects.get(id=1)

*** MAPPING CourseEnrollment to CourseEnrollmentData ***
>>> course_enrollment_data = CourseEnrollmentData(
...                             user=UserData(
...                                 user_pii=(
...                                       username=course_enrollment.user.username,
...                                       email=course_enrollment.user.email,
...                                       name=course_enrollment.user.profile.name                                 
...                                  ),
...                                  id=course_enrollment,ent.user.id,
...                                  is_active=course_enrollment.user.is_active
...                             ),
...                             course=CourseOverviewData(
...                                 course_key=course_enrollment.course.id,
...                                 display_name=course_enrollment.course.display_name
...                             ),
...                             mode=course_enrollment.mode,
...                             creation_date=course_enrollment.created,
...                             is_active=course_enrollment.is_active
...                         )
>>> print(course_enrollment_data)
>>> import attr
>>> attr.asdict(course_enrollment_data)
```

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
**Description:** 
This PR adds class definitions for the objects that will be sent by Open edX events, they are based in the library

**JIRA:**
https://edunext.atlassian.net/browse/PS2021-629?atlOrigin=eyJpIjoiODIwNDA5M2MxZGRlNGQ2NjhjNjJhZWIwNWZiNWNlYjUiLCJwIjoiaiJ9

**Dependencies:** 
This PR depends on #7 

**Testing instructions:**
In your Open edX devstack
1. Go to lms shell `make lms-shell`
2. Install openedx-events specifying the events definition branch:
 `pip install git+https://github.com/eduNEXT/openedx-events.git@MJG/event_data#egg=event_data` 
3. Go to the lms Django shell: `python manage.py lms shell`
4. Now, you can import the platform objects and map them to attrs classes. For example, you could do:

```
>>> from common.djangoapps.student.models import CourseEnrollment
>>> from openedx_events.learning.data import CourseEnrollmentData, UserData, UserPersonalData, CourseOverviewData
>>> course_enrollment = CourseEnrollment.objects.get(id=1)

*** MAPPING CourseEnrollment to CourseEnrollmentData ***
>>> course_enrollment_data = CourseEnrollmentData(
...                             user=UserData(
...                                 user_pii=(
...                                       username=course_enrollment.user.username,
...                                       email=course_enrollment.user.email,
...                                       name=course_enrollment.user.profile.name                                 
...                                  ),
...                                 id=course_enrollment,ent.user.id,
...                                 is_active=course_enrollment.user.is_active
...                             ),
...                             course=CourseOverviewData(
...                                 course_key=course_enrollment.course.id,
...                                 display_name=course_enrollment.course.display_name
...                             ),
...                             mode=course_enrollment.mode,
...                             creation_date=course_enrollment.created,
...                             is_active=course_enrollment.is_active
...                         )
>>> print(course_enrollment_data)
>>> import attr
>>> attr.asdict(course_enrollment_data)
```

**Reviewers:**
- [ ] @felipemontoya 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

**Comment by mariajgrimaldi** [comment]
This PR was opened because #8 close unexpectedly. Please, refer to #8 for design discussions.

File: openedx_events/learning/data.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        name (str): email associated with the Open edX user's profile.
+    """
+
+    username = attr.ib(type=str)
+    email = attr.ib(type=str)
+    name = attr.ib(type=str, factory=str)
+
+
+@attr.s(frozen=True)
+class UserData(UserNonPersonalData):
```
Comment (mariajgrimaldi) [inline openedx_events/learning/data.py:44]: Hey @felipemontoya, thanks for the recommendation you did here.

Comment (felipemontoya) [reply openedx_events/learning/data.py:44]: I think we are

**Review by felipemontoya — APPROVED** [review]
Good work, we are ready to merge this and focus on the usage now.

Btw, the example of code in the cover letter is now outdated.

**Pull request merged**